packages/core/src/entity/promotion/promotion.entity.ts

```diff
@@ -149,6 +158,13 @@ export class Promotion
             const promotionAction = this.allActions[action.code];
             if (promotionAction instanceof PromotionItemAction) {
                 if (this.isOrderItemArg(args)) {
+                    const { orderLine } = args;
+                    amount +=
+                        roundMoney(await promotionAction.execute(ctx, orderLine, action.args, state, this)) *
```

Comment:
the `roundMoney()` function accepts a 2nd argument for quantity, so that the configured MoneyStrategy can control the multiplication.

Reply:
Thank you for the review.
It seems the issue arose because the previous quantity calculation logic was simply transferred.
I have added the calculation logic as an argument to `roundMoney`.

Reply:
Thank you for the review.
It seems the issue arose because the previous quantity calculation logic was simply transferred.
I have added the calculation logic as an argument to `roundMoney`.